app/controllers/admin/customers_controller.rb

```diff
@@ -64,8 +65,13 @@ def destroy
     def collection
       return Customer.where("1=0") unless json_request? && params[:enterprise_id].present?
 
-      enterprise = Enterprise.managed_by(spree_current_user).find_by_id(params[:enterprise_id])
-      Customer.of(enterprise)
+      Customer.of(managed_enterprise_id).
+        includes(:bill_address, :ship_address, user: :credit_cards)
+    end
+
+    def managed_enterprise_id
+      @managed_enterprise_id ||= Enterprise.managed_by(spree_current_user).
+        select('enterprises.id').find_by_id(params[:enterprise_id])
```

Comment:
this select can go to the end of the line, right? I think it will make it easier to understand.

Reply:
I think I tried that previuously and it didn't work, the `#find_by_id` didn't want to take a chained `#select` after it, and complained loudly... :man_shrugging: 

I'll give it another try :+1:

Reply:
Nope, all the tests in customers_controller_spec immediately break with the suggested change, throwing:
```ruby
NoMethodError: private method `select' called for #<Enterprise:0x000055ff6dd0c100>
```

Reply:
ok, thanks for trying. the find_by method doesnt return a relation, it returns the enterprise.